monitoring/monitoring.go

```diff
@@ -460,6 +515,10 @@ func (i *int64Metric) RecordInt(value int64) {
 	stats.Record(i.ctx, i.M(value)) //nolint:errcheck
 }
 
+func (i *int64Metric) disable() {
+	i.disabled = true
+}
+
 func (i *int64Metric) With(labelValues ...LabelValue) Metric {
 	t := make([]tag.Mutator, len(i.tags), len(i.tags)+len(labelValues))
```

Comment:
nit: do we want to make this a NOP for disabled? `with` is pretty slow so that make improve performance?

Reply:
Makes sense. Will change